diff --git a/macism.swift b/macism.swift
index 2791d52..dc3d441 100644
--- a/macism.swift
+++ b/macism.swift
@@ -3,9 +3,6 @@ import Foundation
 @main
 struct MacISM {
     static func main() {
-        // Initialize input sources
-        InputSourceManager.initialize()
-
         if CommandLine.arguments.contains(where: { arg in
             arg.caseInsensitiveCompare("--version") == .orderedSame
         }) {
@@ -13,6 +10,9 @@ struct MacISM {
             return
         }
 
+        // Initialize input sources
+        InputSourceManager.initialize()
+
         if CommandLine.arguments.count == 1 {
             let currentSource = InputSourceManager.getCurrentSource()
             print(currentSource.id)
